storm-core/src/clj/backtype/storm/util.clj

```diff
@@ -392,6 +392,15 @@
       (.addArgument command a))
     (.execute (DefaultExecutor.) command)))
 
+(defn exists-process?
+   [process-id]
+   (let [line (if on-windows? (str "cmd /c \"tasklist /FI \"PID eq "  process-id  "\" | findstr "  process-id  "\"" )
+                              (str "ps -p "  process-id))]
+        (try-cause
+           (exec-command! line)
```

Comment:
I prefer to use /proc/ directly , see:https://github.com/caofangkun/apache-storm/commit/2b2d65095402217d0217272b65ac9e1925125152#diff-60bc01aeb7fe37b1dc8ba418ebd627c3R379
but I am afraid  that  "/proc also does not exist on all unix variants"  as you have mentioned before.